### ISSUE TITLE
Can't connect to Firestore emulator by default with @firebase/testing

### ISSUE THREAD

**Issue by mirkonasato** [firebase/firebase-js-sdk#1759]
### [REQUIRED] Describe your environment

  * Operating System version: macOS 10.14.4
  * Browser version: N/A
  * Firebase SDK version: `"@firebase/testing": "0.9.2"`
  * Firebase Product: Firestore (Emulator)

### [REQUIRED] Describe the problem

After following the [Test your Cloud Firestore Security Rules](https://firebase.google.com/docs/firestore/security/test-rules-emulator) instructions for Node.js, my test script cannot connect to the Firestore Emulator:

```
(node:27279) UnhandledPromiseRejectionWarning: Error: 14 UNAVAILABLE: Connect Failed
    at Object.exports.createStatusError (.../firestore-sample/node_modules/grpc/src/common.js:91:15)
```

#### Steps to reproduce:

Create a new folder `firestore-sample` with the following `package.json` that uses the latest available `firebase-tools` and `@firebase/testing` dependencies:

```json
{
  "name": "firestore-sample",
  "private": true,
  "devDependencies": {
    "@firebase/testing": "^0.9.2",
    "firebase-tools": "^6.9.0"
  }
}
```

and the following `test.js` file:

```javascript
const firebase = require('@firebase/testing');
firebase.clearFirestoreData({projectId: 'test-project'});
```

Start the Firestore Emulator as described in [Test your Cloud Firestore Security Rules](https://firebase.google.com/docs/firestore/security/test-rules-emulator):

```bash
$ $(npm bin)/firebase setup:emulators:firestore
$ $(npm bin)/firebase serve --only firestore
✔  firestore: Emulator running at http://localhost:5002
```

Now running the test script:

```bash
$ node test.js
```

results in this error

```
(node:27279) UnhandledPromiseRejectionWarning: Error: 14 UNAVAILABLE: Connect Failed
    at Object.exports.createStatusError (.../firestore-sample/node_modules/grpc/src/common.js:91:15)
```

Adding some logging to `node_modules/grpc/src/client.js` shows it's trying to connect to `localhost:8080`, but the emulator started on `localhost:5002`.

After some more investigation I found that `node_modules/@firebase/testing/dist/index.cjs.js` contains

```javascript
/** If this environment variable is set, use it for the Firestore emulator. */
var FIRESTORE_ADDRESS_ENV = 'FIREBASE_FIRESTORE_EMULATOR_ADDRESS';
/** The default address for the local Firestore emulator. */
var FIRESTORE_ADDRESS_DEFAULT = 'localhost:8080';
/** The actual address for the Firestore emulator */
var FIRESTORE_ADDRESS = process.env[FIRESTORE_ADDRESS_ENV] || FIRESTORE_ADDRESS_DEFAULT;
```

Surely enough setting that `FIREBASE_FIRESTORE_EMULATOR_ADDRESS` environment variable makes it work:

```bash
$ export FIREBASE_FIRESTORE_EMULATOR_ADDRESS=localhost:5002
$ node test.js
(no errors)
```

Additionally, if I start the emulator using the [gcloud beta emulators firestore](https://cloud.google.com/sdk/gcloud/reference/beta/emulators/firestore/) command (rather than with the `firebase` command installed by `firebase-tools`), it runs on a different port every time, and it suggest setting another environment variable:

```bash
$ gcloud beta emulators firestore start
[firestore] API endpoint: http://::1:8322
[firestore] If you are using a library that supports the FIRESTORE_EMULATOR_HOST environment variable, run:
[firestore]
[firestore]    export FIRESTORE_EMULATOR_HOST=::1:8322
```

but setting `FIRESTORE_EMULATOR_HOST` has no effect with `@firebase/testing`. Again setting `FIREBASE_FIRESTORE_EMULATOR_ADDRESS` does the right thing.

I believe that at a minimum that `FIREBASE_FIRESTORE_EMULATOR_ADDRESS` environment variable should be documented somewhere, and ideally things should work by default without setting any variables.

**Comment by samtstern** [comment]
@mirkonasato thanks for all this feedback!  Seems like you figured out exactly what is happening.  We recently released version `6.9.0` of the `firebase` CLI which has a number of changes to emulators, including new ways of starting them and setting the ports.

First of all we introduced the `emulators:start` command which will create output like this, which warns you about the environment variable:
```
$ firebase emulators:start --only firestore
i  Starting emulators: ["firestore"]
i  firestore: Logging to firestore-debug.log
i  firestore: For testing set FIREBASE_FIRESTORE_EMULATOR_ADDRESS=localhost:5002
```

Secondly it includes a new command `emulators:exec` which helps you to start the emulators, run a command of your choosing, and then stop the emulators.  It runs your command with the environment already configured correctly, which means you could just do:

```
$ firebase emulators:exec --only firestore "node index.js"
```

If I were you I'd wait for `6.9.1` later today before diving into this, we have some known bugs we're fixing as I speak.

**Comment by mirkonasato** [comment]
@samtstern thanks for the quick reply! Just as a suggestion, since sounds like you're in the middle of making changes anyway, why not use the `FIRESTORE_EMULATOR_HOST` variable, since apparently that's what some other SDKs use already? E.g. [go](https://github.com/googleapis/google-cloud-go/blob/master/firestore/client.go#L66) and [ruby](http://googleapis.github.io/google-cloud-ruby/docs/google-cloud-firestore/latest/file.EMULATOR.html). I suspect having different ways to do the same thing for different languages is likely to cause confusion.

**Comment by samtstern** [comment]
We definitely plan to unify the variables, the fact that they differ is
actually a mistake that we didn't catch soon enough.

FYI we're going to change the default ports from 5002 back to 8080 so that
this new emulator command is not a breaking change for you or others.

*From: *Mirko Nasato <notifications@github.com>
*Date: *Tue, May 7, 2019, 9:42 AM
*To: *firebase/firebase-js-sdk
*Cc: *Sam Stern, Mention

@samtstern <https://github.com/samtstern> thanks for the quick reply! Just

**Comment by samtstern** [comment]
The fix for this issue has been released in `6.9.1`.
